Patch nobbled from RedHat, and adapted to work with dhclient and dhrelay
by Roy Marples <uberlord@gentoo.org>

--- dhcp-3.0.3/omapip/errwarn.c.no_isc_blurb	2004-09-30 16:38:32.000000000 -0400
+++ dhcp-3.0.3/omapip/errwarn.c	2005-04-18 13:45:49.037158000 -0400
@@ -45,6 +45,8 @@
 int log_perror = 1;
 #endif
 int log_priority;
+int log_isc_blurb=1;
+
 void (*log_cleanup) (void);
 
 #define CVT_BUF_MAX 1023
@@ -76,7 +78,9 @@
 	  write (STDERR_FILENO, "\n", 1);
   }
 
-#if !defined (NOMINUM)
+#if !defined(NOMINUM)
+  if ( log_isc_blurb )
+  {
   log_error ("%s", "");
   log_error ("If you did not get this software from ftp.isc.org, please");
   log_error ("get the latest from ftp.isc.org and install that before");
@@ -94,7 +98,12 @@
   log_error ("the README file.");
   log_error ("%s", "");
   log_error ("exiting.");
+  }else
+  {
+      log_error ("exiting.");
+  }
 #endif
+
   if (log_cleanup)
 	  (*log_cleanup) ();
   exit (1);
--- dhcp-3.0.3/client/dhclient.c.no_isc_blurb	2005-04-18 13:21:08.509169000 -0400
+++ dhcp-3.0.3/client/dhclient.c	2005-04-18 13:49:27.574402000 -0400
@@ -34,7 +34,7 @@
 static char ocopyright[] =
 "$Id: dhcp-3.0.3-no_isc_blurb.patch,v 1.1 2005/12/09 14:13:30 uberlord Exp $ Copyright (c) 2004 Internet Systems Consortium.  All rights reserved.\n";
 #endif /* not lint */
-
+ 
 #include "dhcpd.h"
 #include "version.h"
 
@@ -81,6 +81,8 @@
 
 void do_release(struct client_state *);
 
+extern int log_isc_blurb;
+
 int main (argc, argv, envp)
 	int argc;
 	char **argv, **envp;
@@ -176,6 +178,7 @@
 		} else if (!strcmp (argv [i], "-q")) {
 			quiet = 1;
 			quiet_interface_discovery = 1;
+			log_isc_blurb = 0;      
 		} else if (!strcmp (argv [i], "-s")) {
 			if (++i == argc)
 				usage ();
--- dhcp-3.0.3/server/dhcpd.c.no_isc_blurb	2005-12-08 10:01:40.000000000 +0000
+++ dhcp-3.0.3/server/dhcpd.c	2005-12-08 10:05:55.000000000 +0000
@@ -62,6 +62,9 @@
 struct iaddr server_identifier;
 int server_identifier_matched;
 
+
+extern int log_isc_blurb;
+
 #if defined (NSUPDATE)
 
 /* This stuff is always executed to figure the default values for certain
@@ -368,6 +371,7 @@
 		} else if (!strcmp (argv [i], "-q")) {
 			quiet = 1;
 			quiet_interface_discovery = 1;
+			log_isc_blurb = 0;
 		} else if (!strcmp (argv [i], "--version")) {
 			log_info ("isc-dhcpd-%s", DHCP_VERSION);
 			exit (0);
--- dhcp-3.0.3/relay/dhcrelay.c.no_isc_blurb	2005-12-08 10:21:00.000000000 +0000
+++ dhcp-3.0.3/relay/dhcrelay.c	2005-12-08 10:21:50.000000000 +0000
@@ -102,6 +102,8 @@
 static char message [] = "Internet Systems Consortium DHCP Relay Agent";
 static char url [] = "For info, please visit http://www.isc.org/sw/dhcp/";
 
+extern int log_isc_blurb;
+
 int main (argc, argv, envp)
 	int argc;
 	char **argv, **envp;
@@ -170,6 +172,7 @@
 		} else if (!strcmp (argv [i], "-q")) {
 			quiet = 1;
 			quiet_interface_discovery = 1;
+			log_isc_blurb = 0;
 		} else if (!strcmp (argv [i], "-a")) {
 			add_agent_options = 1;
 		} else if (!strcmp (argv [i], "-c")) {
